```
/root/.cache/bazel/_bazel_root/ec321eb2cc2d0f8f91b676b6d4c66c29/external/bootlin_i686_toolchain/bin/../lib/gcc/i686-buildroot-linux-gnu/8.1.0/../../../../i686-buildroot-linux-gnu/bin/ld: bazel-out/i686-opt/bin/common/trinary/libtryte_long.a(tryte_long.o): in function `long_to_trytes':
--
  | /proc/self/cwd/common/trinary/tryte_long.c:57: undefined reference to `pow'
```

You're missing parameters like the below from your `CROSSTOOL` files:
```
  cxx_flag: "-std=c++11"
  linker_flag: "-lpthread"
  linker_flag: "-L/usr/local/llvm_scratch/libcxx_msan/lib"
  linker_flag: "-lc++"
  linker_flag: "-lc++abi"
  linker_flag: "-Wl,-rpath,/usr/local/llvm_scratch/libcxx_msan/lib"
  linker_flag: "-lm"
  linker_flag: "-fuse-ld=gold"
  linker_flag: "-B/usr/bin"
  linker_flag: "-B/usr/bin"
  linker_flag: "-Wl,--build-id=md5"
  linker_flag: "-Wl,--hash-style=gnu"
  cxx_builtin_include_directory: "/usr/local/llvm_scratch/libcxx_msan/include/c++/v1"
  cxx_builtin_include_directory: "/usr/local/llvm_scratch/libcxx_msan/include/"
  cxx_builtin_include_directory: "/usr/include/clang/4.0.0/include"
  cxx_builtin_include_directory: "/usr/lib/clang/4.0.0/include"
  cxx_builtin_include_directory: "/usr/lib/llvm-4.0/lib/clang/4.0.0/include"
  cxx_builtin_include_directory: "/usr/local/include"
  cxx_builtin_include_directory: "/usr/include/x86_64-linux-gnu"
  cxx_builtin_include_directory: "/usr/include"
```

--- a/.buildkite/pipeline.yml
+++ b/.buildkite/pipeline.yml
@@ -13,19 +13,3 @@ steps:
         image: "th0br0/iota-buildenv:x86_64"
         mounts:
           - /conf:/conf
-
-  - name: "bazel test opt i686"
-    command: bazel test --crosstool_top=@iota_toolchains//tools/x86-i686--glibc--bleeding-edge-2018.07-1:toolchain --cpu=i686 --host_crosstool_top=@bazel_tools//tools/cpp:toolchain -c opt --test_output=all //...
-    plugins:
-      docker#v1.3.0:
-        image: "th0br0/iota-buildenv:x86_64"
-        mounts:
-          - /conf:/conf
-
-  - name: "bazel test debug i686"
-    command: bazel test --crosstool_top=@iota_toolchains//tools/x86-i686--glibc--bleeding-edge-2018.07-1:toolchain --cpu=i686 --host_crosstool_top=@bazel_tools//tools/cpp:toolchain -c dbg --test_output=all //...
-    plugins:
-      docker#v1.3.0:
-        image: "th0br0/iota-buildenv:x86_64"
-        mounts:
-          - /conf:/conf

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -27,4 +27,5 @@ load("@rules_iota//:defs.bzl", "iota_deps")
 iota_deps()
 
 load("@iota_toolchains//:toolchains.bzl", "setup_toolchains")
+
 setup_toolchains()

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -8,13 +8,7 @@ git_repository(
 
 git_repository(
     name = "iota_toolchains",
-    commit = "984a2e58bce68c2b74100d9fa1f3434e96e92d95",
-    remote = "https://github.com/iotaledger/toolchains.git",
-)
-
-git_repository(
-    name = "iota_toolchains",
-    commit = "a273b913ae18ab80a6e0f5c96e254927ee7c9440",
+    commit = "6b501df8e7f3bc3b143c894737fbb1d82e914762",
     remote = "https://github.com/iotaledger/toolchains.git",
 )
 

--- a/common/trinary/trit_array.h
+++ b/common/trinary/trit_array.h
@@ -102,9 +102,9 @@ static inline uint8_t flex_trit_array_set_at(flex_trit_t *const trit_array,
   uint8_t tindex = index % 5U;
   // Find out the index of the byte in the array
   index = index / 5U;
-  bytes_to_trits(((byte_t *)trit_array + index), 1, trits, 5);
+  bytes_to_trits((byte_t *)(trit_array + index), 1, trits, 5);
   trits[tindex] = trit;
-  trit_array[index] = trits_to_byte(trits, buffer, 5);
+  trit_array[index] = trits_to_byte(trits, 0, 5);
 #endif
   return 1;
 }

